src/opcodes/opcodes.cairo

```diff
@@ -486,4 +487,11 @@ pub mod Opcode {
     fn opcode_1negate(ref engine: Engine) {
         engine.dstack.push_int(-1);
     }
+
+    fn opcode_ifdup(ref engine: Engine) {
+        let a = engine.dstack.peek_byte_array(0);
```

Comment:
It might be better to use the `peek_bool` function here, then you can do a bool if statement below

Reply:
https://github.com/keep-starknet-strange/shinigami/blob/01db74ab012711ab3a96e305cf96c3176819a672/src/stack.cairo#L81

Reply:
Got it!